packages/react-reconciler/src/ReactFiberCompleteWork.new.js

```diff
@@ -687,8 +687,21 @@ function completeWork(
           // It's also safe to do for updates too, because current.child would only be null
           // if the previous render was null (so the the container would already be empty).
           //
-          // The additional root.hydrate check is required for hydration in legacy mode with no fallback.
-          workInProgress.effectTag |= Snapshot;
+          // The additional root.hydrate check above is required for hydration in legacy mode with no fallback.
+          //
+          // The root container check below also avoids a potential legacy mode problem
+          // where unmounting from a container then rendering into it again
+          // can sometimes cause the container to be cleared after the new render.
+          const containerInfo = fiberRoot.containerInfo;
+          const legacyRootContainer =
+            containerInfo == null ? null : containerInfo._reactRootContainer;
```

Comment:
This field is DOM specific. Not sure what happens to this code in RN. I guess it would just schedule a Snapshot which is currently a noop?

Reply:
Yes. RN doesn't have a way to clear its container anyway (since there's no native API).

Reply:
I mentioned this too in my previous comment. It's something we can address in a follow up.

Reply:
Even though it is kind of renderer-specific, I don't know if I agree that moving it into the host config (and commit phase) is a good follow up change. Happy to talk more about it though.

Reply:
Yeah, it sucks either way. I guess a complete phase legacy check is less bad as it has less overhead and we know we'll likely be removing it in the near future.